ui/components/app/simulation-preview/simulation-preview.test.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { SimulationData } from '@metamask/transaction-controller';
+import { Hex } from '@metamask/utils';
+import configureStore from 'redux-mock-store';
+import { renderWithProvider } from '../../../../test/lib/render-helpers';
+
+import mockState from '../../../../test/data/mock-state.json';
+import { Numeric } from '../../../../shared/modules/Numeric';
+import { EtherDenomination } from '../../../../shared/constants/common';
+import { SimulationPreview } from './simulation-preview';
+
+const DUMMY_BALANCE_CHANGE = {
+  previousBalance: '0xIGNORED' as Hex,
+  newBalance: '0xIGNORED' as Hex,
+};
+
+const ONE_ETH = Numeric.from('0xde0b6b3a7640000', 16, EtherDenomination.ETH);
+const HALF_ETH = ONE_ETH.divide(2, 16);
+
+describe('SimulationPreview', () => {
+  const store = configureStore()(mockState);
+
+  it('renders the component with balance changes', () => {
+    const simulationData: SimulationData = {
+      nativeBalanceChange: {
+        ...DUMMY_BALANCE_CHANGE,
+        isDecrease: true,
+        difference: ONE_ETH.toPrefixedHexString() as Hex,
+      },
+      tokenBalanceChanges: [],
+    };
+
+    renderWithProvider(
+      <SimulationPreview simulationData={simulationData} />,
+      store,
+    );
+
+    expect(screen.getByText('You send')).toBeInTheDocument();
```

Comment:
It's a matter of preference but at what point are we asserting enough of the visual that we could instead just use a snapshot test to verify the entire rendering and be notified if it changes?

Reply:
Good point. I will address this in 2nd PR